phone/src/apps/match/hooks/useMatchActions.ts

```diff
@@ -0,0 +1,61 @@
+import { FormattedProfile, MatchEvents, Profile } from '../../../../../typings/match';
+import dayjs from 'dayjs';
+import { useTranslation } from 'react-i18next';
+import { fetchNui } from '../../../utils/fetchNui';
+import { ServerPromiseResp } from '../../../../../typings/common';
+import { useSetFormattedProfiles } from './state';
+import { useSnackbar } from '../../../ui/hooks/useSnackbar';
+import { useCallback } from 'react';
+
+export const useMatchActions = () => {
+  const { t } = useTranslation();
+  const setProfiles = useSetFormattedProfiles();
+  const { addAlert } = useSnackbar();
+
+  const formatProfile = (profile: Profile): FormattedProfile | null => {
+    return {
+      ...profile,
+      tagList: profile.tags.split(',').filter((t) => t), // remove any empty tags
+      lastActiveFormatted: dayjs.unix(profile.lastActive).format(t('DATE_TIME_FORMAT')),
+      viewed: false,
+    };
+  };
+
+  const setViewed = useCallback(
+    (id: number, liked: boolean) => {
+      setProfiles((profiles) =>
+        profiles.map((profile) => {
+          if (id === profile.id) return { ...profile, viewed: true, liked };
+          return profile;
+        }),
+      );
+
+      fetchNui<ServerPromiseResp<boolean>>(MatchEvents.SAVE_LIKES, [{ id, liked }]).then((resp) => {
```

Comment:
Is there any reason we are sending this in an array? That seems a little odd. If possible refactor to standard object.

Reply:
No clue at all. I wanted to ask Kire about it, but didn't get a response, so I left it be.